### PR TITLE
New Transaction Details Scene

### PR DESCRIPTION
**Pull request by amanalili**
Task: Redesign Transaction Details - https://app.asana.com/0/361770107085503/1148627901147581

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [x] Tested on iOS Tablet
- [x] Tested on small Android

### CODE REVIEW

File: src/components/common/TransactionDetailsCategoryInput.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                  )
+                })}
+              </View>
+            </View>
+            <View style={styles.inputSubCategoryContainter}>
+              <FormField
+                autoFocus
+                blurOnSubmit
+                autoCapitalize="none"
+                label="Choose a sub-category"
```
Comment (paullinator) [inline src/components/common/TransactionDetailsCategoryInput.js:96]: Translate

File: src/components/common/TransactionDetailsCategoryInput.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  }
+  onChangeSubCategory = (subCategory: string) => {
+    this.setState({ subCategory })
+    this.props.onChange(this.state.category, subCategory)
+  }
+  onSelectSubCategory = (input: string) => {
+    const { bridge, subCategories, setNewSubcategory } = this.props
+    const splittedSubCategory = input.split(':')
+    const category = splittedSubCategory[0].toLowerCase()
+    const subCategory = splittedSubCategory[1]
```
Comment (paullinator) [inline src/components/common/TransactionDetailsCategoryInput.js:60]: Subcategory is everything after the first : 
More colons may exist but need to be in the subcategory

**Comment by paullinator** [comment]
Please address William and my comments and fix conflicts

File: src/actions/TransactionDetailsActions.js [hunk 3]
```diff
@@ -20,7 +20,6 @@ export const setTransactionDetails = (transaction: EdgeTransaction, edgeMetadata
     .saveTxMetadata(transaction.txid, transaction.currencyCode, edgeMetadata)
     .then(() => {
       dispatch(refreshTransactionsRequest(wallet.id, [transaction]))
-      Actions.pop()
```
[the remark this answers is not in the document]
Comment (amanalili) [reply src/actions/TransactionDetailsActions.js]: This was re-added on following commits. You might be checking older commits.

File: src/constants/IconConstants.js [hunk 4]
```diff
@@ -42,6 +42,7 @@ export const WALLET_ICON = 'ios-wallet'
 export const IOS_INFO_ICON = 'ios-information-circle-outline'
 export const ANDROID_INFO_ICON = 'md-information-circle-outline'
 export const WARNING = 'ios-warning'
+export const CONTACT = 'ios-contact'
```
Comment (paullinator) [inline src/constants/IconConstants.js:45]: Do not use these constants. Import the icon library directly where needed

File: src/locales/en_US.js [hunk 5]
```diff
@@ -398,7 +398,15 @@ const strings = {
   amount_below_limit: 'Transaction amount is below the min limit of %1$s %2$s',
   transaction_details_category_title: 'Category',
   transaction_details_payee: 'Payee',
-  transaction_details_view_advanced_data: 'View advanced transaction data',
+  transaction_details_payer: 'Payer',
+  transaction_details_recepient: 'Recepient',
```
Comment (paullinator) [inline src/locales/en_US.js:402]: Misspelling

File: src/components/common/TransactionDetailsPersonInput.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    return (
+      <AirshipModal bridge={bridge} onCancel={() => bridge.resolve(null)}>
+        <TouchableWithoutFeedback onPress={() => bridge.resolve(null)}>
+          <View style={styles.airshipContainer}>
+            <FormattedText style={styles.airshipHeader}>{personStatusString}</FormattedText>
+            <FormField
+              autoFocus
+              autoCorrect={false}
+              autoCapitalize="words"
+              returnKeyLabel={'done'}
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/components/common/TransactionDetailsPersonInput.js:53]: Still not done

**Comment by swansontec** [comment]
Excellent! @amanalili, I see a few tiny little glitches in this PR that I'll try solving really quick, then I'll merge the whole thing.

**Pull request merged**